crates/hyperqueue/src/client/output/cli.rs

```diff
@@ -530,16 +530,29 @@ impl Output for CliOutput {
             rows.push(vec![state_label, session_to_cell(info.is_open)]);
 
             let mut n_tasks = info.n_tasks.to_string();
-            let ids =
-                IntArray::from_non_overlapping(submit_descs.iter().filter_map(|submit_desc| {
-                    match &submit_desc.task_desc {
-                        JobTaskDescription::Array { ids, .. } => Some(ids),
-                        JobTaskDescription::Graph { .. } => {
-                            // TODO
-                            None
+
+            let ids = if submit_descs.len() == 1
+                && matches!(&submit_descs[0].task_desc, JobTaskDescription::Array { .. })
+            {
+                match &submit_descs[0].task_desc {
+                    JobTaskDescription::Array { ids, .. } => ids.clone(),
+                    _ => unreachable!(),
+                }
+            } else {
+                let mut ids: Vec<u32> = submit_descs
+                    .iter()
+                    .flat_map(|submit_desc| match &submit_desc.task_desc {
+                        JobTaskDescription::Array { ids, .. } => {
```

Comment:
A wanted to use itertools in past, there are more places where it can be utilized so the usage may grow in time. Moreover, our dependencies are already compiling itertools.

Reply:
I see. In that case let's keep it, and feel free to use it wherever you want :)